Scc 3415/m2 delivery locations

charmingduchess opened the pull request:
Adde an m2CustomerCode check to the __deliveryLocationsByCustomerCode function. I didn't see any unit tests for this function, and its a pretty small one, so I didn't write one. But I can! The ticket was estimated at a 5, but this felt more like a 2. Am I missing something?

nonword reviewed (commented):
Looking good. Glad it ended up being so simple. Please do add tests verifying that `__deliveryLocationsByCustomerCode` returns sensible responses for a requestable and no-requestable m2 customer code

charmingduchess commented:
This is combines the work for [3435](https://jira.nypl.org/browse/SCC-3415) [3414.](https://jira.nypl.org/browse/SCC-3414). 

- removed the __ prefix and created two functions out of the single __deliveryLocationsByCustomerCode, one for recap and one for m2
- added NYPL_CORE_VERSION to the travis test script (as long as this code is included in the discovery api and the rom com changes are not in the master branch of nypl-core-objects, compiling this code will fail)

nonword commented on lib/availability_resolver.js:
```diff
@@ -273,14 +273,18 @@ class AvailabilityResolver {
+      if (item.m2CustomerCode && item.m2CustomerCode[0]) {
+        deliveryLocations = DeliveryLocationsResolver.deliveryLocationsByM2CustomerCode(item.m2CustomerCode)
+      }
+      return requestableBasedOnHoldingLocation(item) && deliveryLocations
```
Is it possible for `deliveryLocations` to be an empty array? Should you also assert it's non-empty?

charmingduchess replied:
[The most updated version of this branch](https://github.com/NYPL/discovery-api/blob/SCC-3415/m2-delivery-locations/lib/availability_resolver.js#L288) does make that check. not sure why you're seeing that there.

nonword commented on .travis.yml:
```diff
@@ -1,6 +1,6 @@
 language: node_js
 install: npm install
-script: npm test
+script: NYPL_CORE_VERSION=v1.x-rom-com-prerelease-1.1 npm test
```
There's a risk that application code will depend on this hard-coded value and allow tests to pass and then cause a failure in the deployed app (where the above var is not set). Instead of setting the above var for all tests, can you temporarily set it around the specific test that needs it? e.g.
```
  describe('deliveryLocationsByCustomerCode', () => {
    beforeEach(() => {
      process.env.NYPL_CORE_VERSION = 'v1.x-rom-com-prerelease-1.1'
    })
    afterEach(() => {
      process.env.NYPL_CORE_VERSION = undefined
    })
    ...
```

charmingduchess replied:
that sounds much better. thanks
